### PR TITLE
Refactor HTML utility to streamline tag replacement logic.

### PR DESCRIPTION
**Pull request by ivandrofly**
Simplified the logic for handling invalid and empty `<i>` tags by reducing redundancy and clarifying handling of edge cases. Added dedicated unit tests to ensure the improved functionality works as expected, particularly for scenarios with invalid or misplaced `<i>` tags.

### CODE REVIEW

File: src/Tests/Logic/UtilitiesTest.cs [hunk 1]
```diff
@@ -515,6 +515,19 @@ public void FixInvalidItalicColonBracketItalic()
...
         }
 
+        [TestMethod]
+        [DataRow("Foo<i></i>bar")]
+        [DataRow("Foo<i>    </i>bar")]
+        [DataRow("<i>Foo</i>    <i>bar</i>", "<i>Foobar</i>")]
+        // [DataRow("Foo<b>    </b>bar")]
+        // [DataRow("Foo<u>    </u>bar")]
+        // [DataRow("Foo<font color=\"#808080\"> </font>bar")]
+        // [DataRow("Foo<font color=\"#808080\"></font>bar")]
```
Comment (ivandrofly) [inline src/Tests/Logic/UtilitiesTest.cs:525]: Commented because, ` HtmlUtil.FixInvalidItalicTags(input)` will short circuit if input doesn't contain italic tag

**Comment by ivandrofly** [comment]
This one is ready
